gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/ddl/catalog/GfxdSystemProcedures.java

```diff
@@ -1639,6 +1639,27 @@ public static void CREATE_ALL_BUCKETS(String tableName) throws SQLException {
     }
   }
 
+  public static void REMOVE_METASTORE_ENTRY(String fqtn, Boolean ignoreException) throws SQLException {
+    String schema;
+    String table;
+    int dotIndex;
+    // NULL table name is illegal
+    if (fqtn == null) {
+      throw Util.generateCsSQLException(SQLState.ENTITY_NAME_MISSING);
+    }
+
+    if ((dotIndex = fqtn.indexOf('.')) >= 0) {
+      schema = fqtn.substring(0, dotIndex);
+      table = fqtn.substring(dotIndex + 1);
+    } else {
+      schema = Misc.getDefaultSchemaName(ConnectionUtil.getCurrentLCC());
+      table = fqtn;
+    }
+    ExternalCatalog catalog = Misc.getMemStore().getExistingExternalCatalog();
+    catalog.removeTableUnsafeIfExists(schema, table, ignoreException);
+    CallbackFactoryProvider.getStoreCallbacks().registerCatalogSchemaChange();
```

Comment:
Is this the same as invoking registerCatalogSchema on catalog on above line? Why not just invoke the same on catalog (you may have to typecast it).